src/html5_parser/node.rs

```diff
@@ -592,167 +592,136 @@ mod test {
-            }
-        }
+        let value = node.get_attribute("key").unwrap().unwrap();
+        assert_eq!(value, "value");
     }
 
     #[test]
-    fn test_get_mut_attribute_non_element() {
+    fn get_mut_attribute_non_element() {
         let mut node = Node::new_document();
-        let result = node.get_mut_attribute("key".to_string());
+        let result = node.get_mut_attribute("key");
         assert!(result.is_err());
     }
 
     #[test]
-    fn test_get_mut_attribute() {
+    fn get_mut_attribute() {
         let mut attr = HashMap::new();
         attr.insert("key".to_string(), "value".to_string());
 
         let mut node = Node::new_element("name", attr.clone(), HTML_NAMESPACE);
 
-        if let Ok(value) = node.get_mut_attribute("key".to_string()) {
-            match value {
-                None => assert!(false),
-                Some(value_str) => (*value_str).push_str(" appended"),
-            }
+        let value = node.get_mut_attribute("key").unwrap().unwrap();
+        value.push_str(" appended");
 
-            if let Ok(value2) = node.get_attribute("key".to_string()) {
-                match value2 {
-                    None => assert!(false),
-                    Some(value_str) => assert_eq!(*value_str, "value appended"),
-                }
-            }
-        }
+        let value = node.get_attribute("key").unwrap().unwrap();
+        assert_eq!(value, "value appended");
     }
 
     #[test]
-    fn test_clear_attributes_non_element() {
+    fn clear_attributes_non_element() {
         let mut node = Node::new_document();
         let result = node.clear_attributes();
         assert!(result.is_err());
     }
 
     #[test]
-    fn test_clear_attributes() {
+    fn clear_attributes() {
         let mut attr = HashMap::new();
         attr.insert("key".to_string(), "value".to_string());
 
         let mut node = Node::new_element("name", attr.clone(), HTML_NAMESPACE);
-        if let Ok(_) = node.clear_attributes() {
-            assert_eq!(node.has_attributes(), false);
-        }
+        assert!(node.clear_attributes().is_ok());
+        assert_eq!(node.has_attributes(), false);
     }
 
     #[test]
-    fn test_has_attributes_non_element() {
+    fn has_attributes_non_element() {
         // if node is a non-element, will always return false
         let node = Node::new_document();
         assert_eq!(node.has_attributes(), false);
     }
 
     #[test]
-    fn test_has_attributes() {
+    fn has_attributes() {
         let attr = HashMap::new();
 
         let mut node = Node::new_element("name", attr.clone(), HTML_NAMESPACE);
         assert_eq!(node.has_attributes(), false);
 
-        if let Ok(_) = node.insert_attribute("key".to_string(), "value".to_string()) {
-            assert_eq!(node.has_attributes(), true);
-        }
+        assert!(node.insert_attribute("key", "value").is_ok());
+        assert_eq!(node.has_attributes(), true);
```

Comment:
Here we can have more confidence that the test will fail if we ever get an `Err`.